code/modules/materials/Mat_MaterialProcs.dm

```diff
@@ -738,3 +738,54 @@ triggerOnEntered(var/atom/owner, var/atom/entering)
 		var/list/color = rgb2num(owner.material.getColor())
 		light_c = owner.AddComponent(/datum/component/loctargeting/sm_light, color[1], color[2], color[3], 255 * 0.33)
 		light_c.update(1)
+
+/datum/materialProc/radioactive_temp
+	max_generations = -1
+
+	execute(var/atom/owner, var/temp)
+		if (isobj(owner) && temp > 500 KELVIN)
+			/// How much this individual thing will affect the decay of plasma on its tile. Considers item amounts only for now
+			var/bulkness_multiplier = 1
+			var/obj/O = owner
+			if (isitem(O))
+				var/obj/item/I = O
+				bulkness_multiplier = I.amount
+				if (I.amount <= 0)
+					return // I just have a feeling this will manage to happen and fuck things up
+			if (!O.material.hasProperty("radioactive"))
+				return
+			var/radioactivity = O.material.getProperty("radioactive")
+			var/n_radioactivity = O.material.getProperty("n_radioactive")
+			if (O.material.hasProperty("n_radioactive"))
+				n_radioactivity = O.material.getProperty("n_radioactive")
+			var/turf/T = locate(O.x, O.y, O.z)
+			if (!issimulatedturf(T))
+				return
+			var/turf/simulated/simTurf = T
+			if (!simTurf.air)
+				return
+			if (simTurf.air.toxins > MINIMUM_REACT_QUANTITY)
+				/// Mostly bullshit magic because I don't know how radiation works and plasma isn't real, but is how many moles to convert of existing plasma
+				var/moles_to_convert = (bulkness_multiplier * (1 + radioactivity) * (1 + n_radioactivity) * sqrt(temp)) / 1000
+				if (moles_to_convert > simTurf.air.toxins)
+					simTurf.air.radgas += simTurf.air.toxins
```

Comment:
This needs to handle group processing somehow or it will do Weird Things, see how plasmastone works.